spec/acceptance/splunk_enterprise_spec.rb

```diff
@@ -1,14 +1,22 @@
 require 'spec_helper_acceptance'
 
 describe 'splunk enterprise class' do
+  init = shell('/bin/readlink /sbin/init', acceptable_exit_codes: [0, 1]).stdout
+  service_name = if init.include? 'systemd'
+                   'Splunkd'
+                 else
+                   'splunk'
+                 end
+
   context 'default parameters' do
     # Using puppet_apply as a helper
     it 'works idempotently with no errors' do
       pp = <<-EOS
       class { '::splunk::enterprise': }
       EOS
 
-      # Run it twice and test for idempotency
+      # Run it three times and test for eventual idempotency (Needed for SELINUX limitations)
```

Comment:
can you be more specific about the need, or link to something at least for the poor lass/lad who has no idea what limitation you're referring to?

Reply:
Very fair point.  I've removed this change as it was something I only really needed when testing locally with vagrant_libvirt and it's not otherwise related to this PR.

As for the actual issue.  Puppet updates the selinux attributes on the config files on the second run since it defaults these file [properties](https://puppet.com/docs/puppet/5.4/types/file.html#file-attribute-seluser) to

> the value returned by matchpathcon for the file, if any exists

Puppet doesn't reread these default selinux contexts at any time during a run. Typically, they can be updated by a package being installed (eg splunk).  See https://tickets.puppetlabs.com/browse/PUP-2169

Reply:
Also see https://github.com/voxpupuli/puppet-splunk/pull/153#discussion_r143781103